### ISSUE TITLE
configure KubeControllersConfiguration from operators's chart

### ISSUE THREAD

**Issue by yelhouti** [tigera/operator#1755]
## Expected Behavior
I installed calico using the tigera operator helm chart, I would like to be able to configure KubeControllersConfiguration at install time (more precisely, hostEndpoint: autoCreate)

## Current Behavior
I am not able to do that.

## Possible Solution

- change the chart to allow the configuration of calico-resources for other providers than openshift. (if I understand the code of the operator correctly, if I create a KubeControllersConfiguration there, it config will get merged with the default one).

or 
- add variables to configure the the default KubeControllersConfiguration in the operator.

**Comment by tmjd** [comment]
The helm chart you are referring to is not created from this repo. This request would be more appropriately created against the repo https://github.com/projectcalico/calico.

**Comment by yelhouti** [comment]
@tmjd indeed but it will be relevant there only if:
> if I understand the code of the operator correctly, if I create a KubeControllersConfiguration there, it config will get merged with the default one

Is that the case ?
Also are there variables
> to configure the the default KubeControllersConfiguration in the operator.

The charts can only do what the operator allows it to do, so maybe maybe this should have been formulated as a question here.

**Comment by tmjd** [comment]
The operator does not update/write to KubeControllersConfiguration, it turns on the "main" controllers through environment variables, but AutoHostEndpoints is not a main controller but a "sub-controller" of the node controller (node is always enabled). The kube controller merges the environment variable config and datastore KubeControllersConfiguration so while the main controller is enabled through an Env var the AutoHostEndpoints can be enabled through the datastore.

So tl;dr, it is possible to turn on the AutoHostEndpoints by setting it in the KubeControllersConfiguration.

While I do not think the chart directly allows setting KubeControllersConfiguration, it should be possible to install the helm chart and then as a second step, set the configuration to enable AutoHostEndpoints.
